main.tf

```diff
@@ -0,0 +1,93 @@
+module "logs" {
+  source                   = "git::https://github.com/cloudposse/tf_log_storage.git?ref=master"
+  name                     = "${var.name}"
+  stage                    = "${var.stage}"
+  namespace                = "${var.namespace}"
+  standard_transition_days = "${var.logs_standard_transition_days}"
+  glacier_transition_days  = "${var.logs.glacier_transition_days}"
+  expiration_days          = "${var.logs.expiration_days}"
+}
+
+module "default_label" {
+  source     = "git::https://github.com/cloudposse/tf_label.git?ref=tags/0.2.0"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  name       = "${var.name}"
+  delimiter  = "${var.delimiter}"
+  attributes = ["origin"]
+  tags       = "${var.tags}"
+}
+
+resource "aws_s3_bucket" "default" {
+  bucket        = "${var.hostname}"
+  acl           = "public-read"
+  tags          = "${module.default_label.tags}"
+  region        = "${var.region}"
+  force_destroy = "${var.force_destroy}"
+
+  logging {
+    target_bucket = "${module.logs.bucket_id}"
+    target_prefix = "${module.logs.prefix}"
+  }
+
+  website {
+    index_document           = "${var.index_document}"
+    error_document           = "${var.error_document}"
+    redirect_all_requests_to = "${var.redirect_all_requests_to}"
+  }
+
+  cors_rule {
+    allowed_headers = "${var.cors_allowed_headers}"
+    allowed_methods = "${var.cors_allowed_methods}"
+    allowed_origins = "${var.cors_allowed_origins}"
+    expose_headers  = "${var.cors_expose_headers}"
+    max_age_seconds = "${var.cors_max_age_seconds}"
+  }
+
+  versioning {
+    enabled = "${var.versioning_enabled}"
+  }
+
+  lifecycle_rule {
```

Comment:
I don't sure about `lifecycle_rule ` for website bucket. Could you explain for what it need?

Reply:
This is so we can have versioned content in the bucket. The lifecycle rule manages how long we keep those old versions.